tcms/telemetry/api.py

```diff
@@ -169,3 +169,59 @@ def _append_status_counts_to_result(count, result):
     for status in TestExecutionStatus.chart_status_names:
         status_count = count.get(status, 0)
         result.get(status).append(status_count)
+
+
+@rpc_method(name='Testing.test_case_health')
+def test_case_health(query=None):
+
+    if query is None:
+        query = {}
+
+    all_test_executions = TestExecution.objects.filter(**query)
+
+    test_executions = _get_count_for(all_test_executions)
+    passed_test_executions = _get_count_for(all_test_executions.filter(
+        status__name__in=TestExecutionStatus.passed_status_names))
+    failed_test_executions = _get_count_for(all_test_executions.filter(
+        status__name__in=TestExecutionStatus.failure_status_names))
+
+    data = {}
+    for te in test_executions:
+        data[te['case_id']] = {
+            'case_id': te['case_id'],
+            'case_summary': te['case__summary'],
+            'count': {
+                'all': te['count'],
+                'success': 0,
+                'fail': 0
+            }
+        }
+
+    for te in passed_test_executions:
+        data[te['case_id']]['count']['success'] = te['count']
+    for te in failed_test_executions:
+        data[te['case_id']]['count']['fail'] = te['count']
+
+    for key in data.copy():
```

Comment:
Use `dict.fromkeys(data)` to avoid possibly copying a very large data structure. This will return new dictionary with the same keys where the values are `None`.